central/graphql/resolvers/node_scan.go

```diff
@@ -29,5 +42,52 @@ func (resolver *nodeScanResolver) NodeComponentCount(_ context.Context, args Raw
 }
 
 func (resolver *nodeScanResolver) NodeComponents(_ context.Context, args PaginatedQuery) ([]NodeComponentResolver, error) {
```

Comment:
Do we need the:
```go
if resolver.ctx == nil {
resolver.ctx = ctx
}
```

Reply:
nodeScanResolver is created by `Scan()` sub-resolver is nodes graphQL and that sub-resolver adds the node's scope to nodeScanResolver's ctx before returning. So, nodeScanResolver should have a non-nil ctx at this point. But I think it is safe to add the check here in case the `Scan` sub-resolver changes.